pytorch_lightning/trainer/trainer.py

```diff
@@ -1147,9 +1151,19 @@ def __load_ckpt_weights(self, ckpt_path: Optional[str]) -> Optional[str]:
         if not self._device_type == DeviceType.TPU:
             self.training_type_plugin.barrier()
 
-        self.training_type_plugin.restore_model_state_from_ckpt_path(
-            ckpt_path, map_location=lambda storage, loc: storage
-        )
+        # Serialize checkpoint loading to avoid OOMs
+        if self.serialize_checkpoint_loading and self.num_gpus > 0:
```

Comment:
Why not let this responsibility to the training_type_plugin, I guess this is useful only for DDP and derivate right now.